test/e2e/network_test.go

```diff
@@ -148,6 +153,33 @@ func testEastWestNetworking(t *testing.T) {
 	}
 }
 
+// getDeploymentLogs ensures logs of multiple deployments
+// are collected via a pod labelSelector
+func (tc *testContext) getDeploymentLogs(deployments ...*appsv1.Deployment) {
+	for _, deployment := range deployments {
+		labelSelector := deployment.Name
+		tc.savePodLogs(labelSelector)
+	}
+}
+
+// getLogs gets the logs for a job or deployment pod specified by the pod labelSelector
+func (tc *testContext) getLogs(podLabelSelector string) (string, error) {
+	if strings.HasPrefix(podLabelSelector, "job-name=") {
+		cmd := exec.Command("oc", "logs", "-l="+podLabelSelector, "-n=wmco-test")
+		logStream, err := cmd.Output()
+		if err != nil {
+			return "", errors.Errorf("error reading pod logs")
+		}
+		return string(logStream), nil
```

Comment:
you could add an if else to only select the command to be executed between each case while keeping the code from these lines to be executed only once. This will reduced the repeated code to get the logStream output and return it.

Reply:
+1

Reply:
Thanks. that's much simpler